p2p/server.go

```diff
@@ -458,6 +469,16 @@ running:
 			// it will keep the node connected.
 			glog.V(logger.Detail).Infoln("<-addstatic:", n)
 			dialstate.addStatic(n)
+		case n := <-srv.removestatic:
+			// This channel is used by RemovePeer to send a
+			// disconnect request to a peer and begin the
+			// stop keeping the node connected
+			glog.V(logger.Detail).Infoln("<-removestatic:", n)
+			p := peers[n.ID]
+			if p != nil {
+				p.disc <- DiscRequested
+				dialstate.removeStatic(n)
+			}
```

Comment:
You're right about blocking, this would be fixed by calling this function instead
```go
func (p *Peer) Disconnect(reason DiscReason)
```

is a rebase preferable for updating this pull-request?